npm/extensions.bzl

```diff
@@ -134,6 +134,9 @@ WARNING: Cannot determine home directory in order to load home `.npmrc` file in
     system_tar = detect_system_tar(module_ctx)
 
     for i in imports:
+        link_packages = {}
+        for link_package, link_names in i.link_packages.items():
+            link_packages[link_package] = [link_name["pkg"] for link_name in link_names]
         npm_import(
```

Comment:
(this is probably beyond this scope of this PR, but I'd like to start the discussion here...)

I think the `npm_import(dev)` flag is flawed and should actually be removed completely. The dev vs prod logic needs to be 100% at package link time based on the package.json dev vs non-dev dependencies, not the lockfile `dev` flag which is currently misused in rules_js (and non-existent in pnpm9).

Does that sound right to you?

Reply:
I agree that the behavior of lockfile `dev` flag makes it unsuitable for really anything that rules_js wants to do with regard to dev dependencies.